src/chatbot/run_chatbot.py

```diff
@@ -45,17 +57,49 @@ def modify_response(response: str, history_len: int) -> str:
     response = response.partition('\n')[0]
     # Remove everything after the last '.', '!', or '?'.
     last_sentence_end_index = max(response.rfind('.'), response.rfind('?'), response.rfind('!'))
-    if last_sentence_end_index != -1:   # -1 means none were found
-        response = response[:last_sentence_end_index+1]
+    if last_sentence_end_index != -1:  # -1 means none were found
+        response = response[:last_sentence_end_index + 1]
     return response.strip()
 
 
-def get_best_response(responses_list: list[str], history_len: int) -> str:
-    # TODO
-    return modify_response(responses_list[0], history_len)
+def get_best_response(responses_list: list[str], history_len: int, prompt_embedding: np.array) -> str:
+    responses = list(map(lambda x: modify_response(x, history_len), responses_list))
```

Comment:
Wouldn't responses = [modify_response(response, history_len) for response in responses_list] do the same thing?

Reply:
It would, but don't you like functional programming? :pleading_face: